cmd/envtool/tests.go

```diff
@@ -448,6 +433,71 @@ func listTestFuncs(dir string) ([]string, error) {
 	return res, nil
 }
 
+// listTestFuncsWithRegex returns regex-filtered names of all top-level test
+// functions (tests, benchmarks, examples, fuzz functions) in the specified
+// directory and subdirectories.
+func listTestFuncsWithRegex(dir, run, skip string) ([]string, error) {
+	tests, err := listTestFuncs(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	if run == "" && skip == "" {
+		return tests, nil
+	}
+
+	includeRegex, err := regexp.Compile(run)
+	if err != nil {
+		return nil, err
+	}
+
+	if skip == "" {
+		return filterStringsByRegex(tests, includeRegex, nil), nil
+	}
+
+	excludeRegex, err := regexp.Compile(skip)
+	if err != nil {
+		return nil, err
+	}
+
+	return filterStringsByRegex(tests, includeRegex, excludeRegex), nil
+}
+
+func filterStringsByRegex(tests []string, include, exclude *regexp.Regexp) []string {
```

Comment:
Could we have documentation on unexported functions as well :hugs:  https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md#code-style-and-conventions

Reply:
Looks resolved to me!